pkg/utils/utils_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"testing"
+)
+
+func TestValidateContainerID(t *testing.T) {
+	testData := []struct {
+		description string
+		containerID string
+		err         bool
+	}{
+		{
+			description: "empty containerID",
+			containerID: "",
+			err:         true,
+		},
+		{
+			description: "invalid characters in containerID",
+			containerID: "1234%%%",
+			err:         true,
+		},
+		{
+			description: "normal containerID",
+			containerID: "a51debf7e1eb",
+			err:         false,
+		},
+	}
+
+	for _, tt := range testData {
+		err := ValidateContainerID(tt.containerID)
+		if err != nil && !tt.err {
```

Comment:
Could you help to use DeepEqual to squash these lines?

Reply:
They are different structure types, and it seems that it is not suitable to use DeepEqual. If they are changed to agree types and then compared, more code will be generated.

Reply:
FYI https://play.golang.org/p/DicOkjUajmv

Reply:
updated, thanks for your suggestions. PTAL